leaderboard/enriching/enricher.go

```diff
@@ -30,6 +31,15 @@ type (
 
 		// WebhookTimeout is the timeout for the webhook call.
 		WebhookTimeout time.Duration `mapstructure:"webhook_timeout,default=2s"`
+
+		Cache *Cache `mapstructure:"cache"`
+	}
+
+	Cache struct {
+		database.RedisOptions
```

Comment:
This seems to me that you are coupling stuff. Enrichment should have its own Redis configurations, maybe allowing only a single instance instead of having cluster support at first.

Moreover, we usually put structures meant to map configurations in the `config/config.go` and each domain/package defines its own configurations. The way you are doing is coupling things too. `config.go` should not depend on anything from the domain code and the same goes for the other way around.

`config.go` is meant to serve configurations for domains in a non-coupling way, as a sort of orchestrator/manager.

You should move `EnrichOptions` and the newly created Redis options to `config.go` and make `Enricher` have its own public configurations. At first it seems like a lot of work, but it makes the code more maintainable and uncoupled. If you need, `api` or event `config` can provide mappers from `config`'s structs to `Enricher`'s structs/options (if you choose the functional option pattern).

Reply:
Refactor done. Thanks for poiting it out.